scripts/apply_freezer_location.py

```diff
@@ -0,0 +1,35 @@
+#!/opt/gls/clarity/users/glsai/Applications/clarity_scripts/UserPreparedLibrary20180523/bin/python3.4
+import sys
+
+from EPPs.common import StepEPP, step_argparser
+
+
+class ApplyFreezerLocation(StepEPP):
+    """
+    Updates submitted sample UDFs for Freezer and Shelf location based on the values provided in step UDFs Freezer and Shelf
+    Sytem exit occurs if step UDF container does not match a sample artificat container name
+    """
+
+    def _run(self):
+
+        for art in self.artifacts:
+            sample = art.samples[0]
+            if sample.artifact.container.name == self.process.udf['Container Name']:
+                sample.udf['Freezer'] = self.process.udf['Freezer']
+                sample.udf['Shelf'] = self.process.udf['Shelf']
+                sample.put()
```

Comment:
There is a `samples` property of `StepEPP` that you should use. it retrieves the samples associated with the input artifacts.
You should also make use of the batch update to make it faster.
That should look like
```python
samples_to_update = set()
for sample in self.samples:
    if sample.artifact.container.name == self.process.udf['Container Name']:
        sample.udf['Freezer'] = self.process.udf['Freezer']
        sample.udf['Shelf'] = self.process.udf['Shelf']
        samples_to_update.add(sample)
    else:
        print("Container '%s' not present in step" % self.process.udf['Container Name'])
        sys.exit(1)
self.lims.put_batch(list(samples_to_update))
```
It will also avoid stopping half way through the script with part of the samples uploaded and the others not